packages/@react-aria/toast/src/useToast.ts

```diff
@@ -83,19 +86,55 @@ export function useToast<T>(props: AriaToastProps<T>, state: ToastState<T>, ref:
     };
   }, [ref]);
 
+  // To improve screen reader announcement of the toast,
+  // we use aria-hidden="true" to hide the toast content
+  // briefly while it is animating in.
+  const [isBusy, setIsBusy] = useState(true);
+  useLayoutEffect(() => {
+    let timeoutId:ReturnType<typeof setTimeout>;
+    if (isBusy && ref.current) {
+      timeoutId = setTimeout(() => setIsBusy(false), 50);
+    }
+    return () => {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+    };
+  }, [ref, isBusy]);
+
   let titleId = useId();
   let descriptionId = useSlotId();
   let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-aria/toast');
 
+  let onClosePress = () => {
+    state.close(key);
+    // Only move focus when there is another Toast. At this point,
+    // state.visibleToasts still includes Toast being removed.
+    if (state.visibleToasts?.length > 1) {
+      const from = document.activeElement?.closest('[role="alertdialog"]') || document.activeElement;
+      const accept = (node:Element) => node.getAttribute('role') === 'alertdialog';
+      let nextItemFocused = focusManager.focusNext({from, accept});
+      if (!nextItemFocused || Object.keys(nextItemFocused).length === 0) {
+        focusManager.focusPrevious({from, accept});
+      }
+    }
+  };
+
   return {
     toastProps: {
-      role: 'alert',
+      role: 'alertdialog',
       'aria-label': props['aria-label'],
       'aria-labelledby': props['aria-labelledby'] || titleId,
       'aria-describedby': props['aria-describedby'] || descriptionId,
       'aria-details': props['aria-details'],
       // Hide toasts that are animating out so VoiceOver doesn't announce them.
-      'aria-hidden': animation === 'exiting' ? 'true' : undefined
+      'aria-hidden': animation === 'exiting' ? 'true' : undefined,
+      tabIndex: 0
+    },
+    contentProps: {
```

Comment:
we'll need to keep this in mind for announcement purposes, it might be argued that it's a breaking change